domain/consensus/finality_test.go

```diff
@@ -167,3 +172,248 @@ func TestFinality(t *testing.T) {
 		}
 	})
 }
+
+func TestBoundedMergeDepth(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		// Set finalityInterval to 50 blocks, so that test runs quickly
+		params.FinalityDuration = 50 * params.TargetTimePerBlock
```

Comment:
Please copy the params before modifying them to avoid possibly interfering with other tests.

Reply:
That sounds like the job of `testutils.ForAllNets` not each individual test